stac_fastapi/api/stac_fastapi/api/app.py

```diff
@@ -79,100 +84,200 @@ def get_extension(self, extension: Type[ApiExtension]) -> Optional[ApiExtension]
+            ),
+        )
+
+    def register_get_search(self):
+        """Register search endpoint (GET /search).
+
+        Returns:
+            None
+        """
+        fields_ext = self.get_extension(FieldsExtension)
+        self.router.add_api_route(
             name="Search",
             path="/search",
-            response_model=ItemCollection if not fields_ext else None,
+            response_model=(ItemCollection if not fields_ext else None)
+            if self.settings.enable_response_models
+            else None,
+            response_class=self.response_class,
             response_model_exclude_unset=True,
             response_model_exclude_none=True,
             methods=["GET"],
-            endpoint=create_endpoint(self.client.get_search, SearchGetRequest),
+            endpoint=self._create_endpoint(self.client.get_search, SearchGetRequest),
         )
-        router.add_api_route(
+
+    def register_get_collections(self):
+        """Register get collections endpoint (GET /collections).
+
+        Returns:
+            None
+        """
+        self.router.add_api_route(
             name="Get Collections",
             path="/collections",
-            response_model=List[Collection],
+            response_model=List[Collection]
+            if self.settings.enable_response_models
+            else None,
+            response_class=self.response_class,
             response_model_exclude_unset=True,
             response_model_exclude_none=True,
             methods=["GET"],
-            endpoint=create_endpoint(self.client.all_collections, EmptyRequest),
+            endpoint=self._create_endpoint(self.client.all_collections, EmptyRequest),
         )
-        router.add_api_route(
+
+    def register_get_collection(self):
+        """Register get collection endpoint (GET /collection/{collectionId}).
+
+        Returns:
+            None
+        """
+        self.router.add_api_route(
             name="Get Collection",
             path="/collections/{collectionId}",
-            response_model=Collection,
+            response_model=Collection if self.settings.enable_response_models else None,
+            response_class=self.response_class,
             response_model_exclude_unset=True,
             response_model_exclude_none=True,
             methods=["GET"],
-            endpoint=create_endpoint(self.client.get_collection, CollectionUri),
+            endpoint=self._create_endpoint(self.client.get_collection, CollectionUri),
         )
-        router.add_api_route(
+
+    def register_get_item_collection(self):
+        """Register get item collection endpoint (GET /collection/{collectionId}/items).
+
+        Returns:
+            None
+        """
+        self.router.add_api_route(
             name="Get ItemCollection",
             path="/collections/{collectionId}/items",
-            response_model=ItemCollection,
+            response_model=ItemCollection
```

Comment:
Would be a nice feature to override the pydantic response_model via config like [this draft PR was doing](https://github.com/stac-utils/stac-fastapi/pull/134/files#diff-4f8c5fdc9b3cca78110705589c68a6df26ac7a9d66bd4af3108db5174fdfe49cR139), but I think this can be done in a follow-up. Reminder to self to create an issue.